### ISSUE TITLE
Dockerfile example with conda create fails for packages from conda-forge

### ISSUE THREAD

**Issue by matthewfeickert** [conda/conda-lock#266]
The Dockerfile example from the README

https://github.com/conda-incubator/conda-lock/blob/04e24386914158e10f6c60084655fdf207df008d/README.md?plain=1#L382-L410

shows using 

```
conda create -p /opt/env --copy --file <lockfile path>
```

but this fails with

```
could not parse '- url: conda-forge' in: /docker/conda-lock.yml

CondaValueError: could not parse '- url: conda-forge' in: /docker/conda-lock.yml
```

if the packages in the lock file are from the conda-forge channel.

## Minimal Failing Example

 * `environment.yml`
```yaml
name: base
channels:
  - conda-forge
dependencies:
  - python==3.8.*
  - awkward==1.10.1
  - pip>=22.3
  - pip:
    - fastjet==3.3.4.0rc9
```

* `Dockerfile`

```dockerfile
FROM condaforge/mambaforge:4.14.0-0 as builder

COPY environment.yml /docker/environment.yml

RUN . /opt/conda/etc/profile.d/conda.sh && \
    mamba create --name lock && \
    conda activate lock && \
    mamba env list && \
    mamba install --yes pip && \
    echo "conda-lock fails to lock if installed with mamba" && \
    echo "c.f. https://github.com/conda-incubator/conda-lock/issues/265" && \
    python -m pip install --upgrade pip setuptools wheel && \
    python -m pip install 'conda-lock==1.2.1' && \
    conda-lock lock \
        --platform linux-64 \
        --file /docker/environment.yml \
        --kind lock \
        --lockfile /docker/conda-lock.yml

RUN mamba create --prefix /opt/env --copy --file /docker/conda-lock.yml

FROM condaforge/mambaforge:4.14.0-0 as base

COPY --from=builder /opt/env /opt/env
```

```
$ docker pull condaforge/mambaforge:4.14.0-0
$ docker build --file Dockerfile --tag conda-lock/debug:issue-266 . &> issue-266-fail.log
```

[issue-266-fail.log](https://github.com/conda-incubator/conda-lock/files/9873805/issue-266-fail.log)

## Minimal Working Example with `conda-lock install`

```dockerfile
FROM condaforge/mambaforge:4.14.0-0 as builder

COPY environment.yml /docker/environment.yml

RUN . /opt/conda/etc/profile.d/conda.sh && \
    mamba create --name lock && \
    conda activate lock && \
    mamba env list && \
    mamba install --yes pip && \
    echo "conda-lock fails to lock if installed with mamba" && \
    echo "c.f. https://github.com/conda-incubator/conda-lock/issues/265" && \
    python -m pip install --upgrade pip setuptools wheel && \
    python -m pip install 'conda-lock==1.2.1' && \
    conda-lock lock \
        --platform linux-64 \
        --file /docker/environment.yml \
        --kind lock \
        --lockfile /docker/conda-lock.yml

RUN . /opt/conda/etc/profile.d/conda.sh && \
    conda activate lock && \
    conda-lock install \
        --mamba \
        --prefix /opt/env \
        /docker/conda-lock.yml

FROM condaforge/mambaforge:4.14.0-0 as base

COPY --from=builder /opt/env /opt/env
```

```console
$ docker run --rm -ti conda-lock/debug:issue-266
(base) root@90fa9c3f4694:/# conda activate /opt/env
(/opt/env) root@90fa9c3f4694:/# python -m pip show awkward
Name: awkward
Version: 1.10.1
Summary: Manipulate JSON-like data with NumPy-like idioms.
Home-page: https://github.com/scikit-hep/awkward-1.0
Author: Jim Pivarski
Author-email: pivarski@princeton.edu
License: BSD-3-Clause
Location: /opt/env/lib/python3.8/site-packages
Requires: importlib-resources, numpy, packaging
Required-by: fastjet
(/opt/env) root@90fa9c3f4694:/# 
```

**Comment by matthewfeickert** [comment]
As a follow up to this, if I wanted to try to copy out the created virtual environment to a new image and then actually use that as a default base environment, would I need to use `conda-pack`? Or is there a feasible way to do this?

**Comment by mariusvniekerk** [comment]
> As a follow up to this, if I wanted to try to copy out the created virtual environment to a new image and then actually use that as a default base environment, would I need to use `conda-pack`? Or is there a feasible way to do this?

So in order for that to work nicely we need to support the `--copy` flag in the install part and then it can be copied over to the exact same location.  Let me take a look at fixing that part.

**Comment by mariusvniekerk** [comment]
This should be addressed in #267

**Comment by constant-roussel** [comment]
@mariusvniekerk any update on this ? I try to build my docker image using https://conda.github.io/conda-lock/docker/ tutorial, but I still have `CondaValueError: could not parse '- url: conda-forge'`

conda-lock, version 1.4.0

**Comment by maresb** [comment]
@constant-roussel, based on the error mesage it looks like you're using a new-style `conda-lock.yml` while `conda-lock` is expecting an old-style explicit `conda-linux-64.lock` file.

The command in the docs to generate the explicit lockfile was wrong, so I proposed a fix in #335. Does that solve the issue? If not, could you please be more specific regarding the Dockerfile you're using, and the commands for generating the lockfile?
